### DISCUSSION TITLE
TIFF Encoder support saving grayscale using 16 bits (L16)

### DISCUSSION THREAD

**Discussion by ptasev** [SixLabors/ImageSharp#2333] [Ideas] [answered]
We have a large grayscale image (1 gigapixels) of the L16 pixel format. Currently it appears the TIFF encoder does not support saving 16 bits per sample and 1 sample per pixel images. At the moment this setup is saved as 8 bits per sample and 3 samples per pixel.

https://github.com/SixLabors/ImageSharp/blob/03cc807eb59c36575a875e3f1c82418b2ba74410/src/ImageSharp/Formats/Tiff/TiffBitsPerPixel.cs#L54-L59

https://github.com/SixLabors/ImageSharp/blob/03cc807eb59c36575a875e3f1c82418b2ba74410/src/ImageSharp/Formats/Tiff/TiffEncoderCore.cs#L376-L390

This feature would allow us to properly save the image as grayscale and not use an extra 8-bits per pixel when using the following encoder settings.
```csharp
var image = new Image<L16>(4, 2);
image.SaveAsTiff("tmp.tiff", new TiffEncoder() { BitsPerPixel = TiffBitsPerPixel.Bit16, PhotometricInterpretation = TiffPhotometricInterpretation.BlackIsZero});
```

From doing a quick look through the code on GitHub, it should be fairly straightforward to implement this. I can give implementing this a shot if the team would like. Ideally it can go in both 2.1.X and v3.

**Comment by JimBobSquarePants** [comment, marked as answer]
Hi @ptasev 

It would be awesome if you could implement this, thanks!

We wouldn't backport a feature into 2.1.x though only critical (i.e. security driven) defects. V3 is very close to completion and will be released hopefully within the next few weeks.

**Reply by ptasev** [reply]
Added in PR https://github.com/SixLabors/ImageSharp/pull/2339
